daemon/main.go

```diff
@@ -31,8 +32,9 @@ func isDpuMode(mode string) (bool, error) {
 
 func createDaemon(dpuMode bool) (Daemon, error) {
 	plugin := plugin.NewGrpcPlugin(dpuMode)
+	dp := deviceplugin.NewGrpcPlugin()
 	if dpuMode {
-		return NewDpuDaemon(plugin), nil
+		return NewDpuDaemon(plugin, dp), nil
```

Comment:
I like how you dependency-inject the dp.

Reply:
it was a necessity :)